azkaban-common/src/main/java/azkaban/executor/FetchActiveFlowDao.java

```diff
@@ -53,7 +53,7 @@ public FetchActiveFlowDao(final DatabaseOperator dbOperator) {
     }
   }
 
-  private static class FetchActiveExecutableFlows implements
+  static class FetchActiveExecutableFlows implements
```

Comment:
consider adding @VisibleForTesting ?

Reply:
Do you have some grounds for using `@VisibleForTesting`? Is it for example checked automatically in the build that it's respected ie. not accessed by other than test code?

Otherwise it seems like redundant documentation that could easily get outdated and thus isn't too useful. If the wider visibility of the method is later desired also for some usage in non-test code, one might easily miss removing the `@VisibleForTesting` annotation and just use the method.

It's easy enough to find references if you want to check where the method is actually used. And if the annotation can't be trusted to be up-to-date, you'd need to check the actual refs any way.

As long as there isn't static code analysis that makes the build fail if `@VisibleForTesting` is not respected, I wouldn't use the annotation at all.

Just my thoughts. If you want that annotation added to get this merged, I'd be happy to do it. Please let me know.

Reply:
Thanks for sharing your thought. 
No we don't have any build tool respecting the annotation, 
```The @VisibleForTesting annotation advertises to other developers (or reminds the code's author) when a decision was made for relaxed visibility to make testing possible or easier```
https://dzone.com/articles/two-generally-useful-guava

This annotation is for reminder purpose only, just like a line of comment, and the only reason here to relax the access level is for testing purpose, so I would imagine that the annotation would be helpful for other people who read this piece of code to understand why the inner class is not private, which worths explanation. I agree there's a maintenance cost as you said, but there's no other simpler alternative than it to let other developers know why we made such decision.

Reply:
Ok, I will add `@VisibleForTesting`.

> there's no other simpler alternative than it to let other developers know why we made such decision

You can search for actual current references and deduce that the wider visibility is for tests (if it's still the case). Additionally, git history can be used to find out the exact (original) intention. These "built-in" ways are _not as fast_ as reading an annotation, but they can be _trusted_ because they never get outdated like a manually maintained explicit annotation might do. Which one is _simpler_, that's a matter of taste I guess :)